cmd/connector/client/util.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2023, SailPoint Technologies, Inc. All rights reserved.
+package connclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+const (
+	ResponseTypeOutput = "output"
+	ResponseTypeState  = "state"
+)
+
+// RawResponse represents the response format from the connector
+type RawResponse struct {
+	Type string          `json:"type"`
+	Data json.RawMessage `json:"data"`
+}
+
+// parseResponse parses a single response into the new format if the given response is in deprecated format
+func parseResponse(resp []byte) (rawResp *RawResponse, err error) {
+	rawResps, _, err := parseResponseList(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(rawResps) != 1 {
+		return nil, fmt.Errorf("failed to parse response")
+	}
+
+	return &rawResps[0], nil
+}
+
+// parseResponseList parses responses into the new format if the given response is in deprecated format
+func parseResponseList(resp []byte) (rawResps []RawResponse, state *RawResponse, err error) {
+	decoder := json.NewDecoder(bytes.NewReader(resp))
+	deprecatedFormat := false
+	for {
+		rr := &RawResponse{}
+		err = decoder.Decode(rr)
```

Comment:
nit:since byte slice, we can unmarshall here.

Reply:
Since this is used for both account and entitlement, we have to defer the unmarshalling process to the upstream caller